### PR TITLE
Make preflight tests faster on GitHub and run them on PRs

### PR DESCRIPTION
**Pull request by kzys**
This change makes preflight tests faster on GitHub by running 5 VMs in parallel and skipping few slow tests.

### CODE REVIEW

File: .github/workflows/preflight.yml
Comment (tvdfly) [inline .github/workflows/preflight.yml:46]: will you move this run script this into a real script or a make target? debugging and editing stuff in the github workflow yaml is... challenging :-)

Comment (kzys) [reply .github/workflows/preflight.yml:46]: Haha. Yeah. Let me extract that as a small shell script.

**Review by tvdfly — APPROVED** [review]
lgtm, thx!

What will be the behavior for PRs opened from forks? Is there a way to get those to run if a committer approves?

**Comment by kzys** [comment]
> Is there a way to get those to run if a committer approves?

GitHub asks maintainers to approve PRs' actions from first-time contributors. We could figure out the way once we have bad actors repeatedly. For example, containerd's Kubenertes tests always need approvals before running.

https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/

**Pull request merged**
